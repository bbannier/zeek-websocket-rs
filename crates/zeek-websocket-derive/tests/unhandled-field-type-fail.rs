use std::collections::BTreeSet;
use zeek_websocket_derive::ZeekType;

#[derive(ZeekType)]
struct X {
    xs: BTreeSet<u8>,
}

fn main() {}
