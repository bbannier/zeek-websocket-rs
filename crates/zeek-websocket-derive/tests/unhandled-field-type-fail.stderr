error[E0277]: the trait bound `Value: From<BTreeSet<u8>>` is not satisfied
 --> tests/unhandled-field-type-fail.rs:4:10
  |
4 | #[derive(ZeekType)]
  |          ^^^^^^^^ the trait `From<BTreeSet<u8>>` is not implemented for `Value`
  |
  = help: the following other types implement trait `From<T>`:
            `Value` implements `From<&str>`
            `Value` implements `From<()>`
            `Value` implements `From<Data>`
            `Value` implements `From<DateTime>`
            `Value` implements `From<HashMap<K, V>>`
            `Value` implements `From<HashSet<T>>`
            `Value` implements `From<IpAddr>`
            `Value` implements `From<IpNetwork>`
          and $N others
  = note: this error originates in the derive macro `ZeekType` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0271]: type mismatch resolving `<BTreeSet<u8> as TryFrom<Value>>::Error == ConversionError`
 --> tests/unhandled-field-type-fail.rs:4:10
  |
4 | #[derive(ZeekType)]
  |          ^^^^^^^^ expected `ConversionError`, found `Infallible`
  |
  = note: this error originates in the derive macro `ZeekType` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `BTreeSet<u8>: TryFrom<Value>` is not satisfied
 --> tests/unhandled-field-type-fail.rs:4:10
  |
4 | #[derive(ZeekType)]
  |          ^^^^^^^^ the trait `From<Value>` is not implemented for `BTreeSet<u8>`
  |
  = help: the trait `From<Value>` is not implemented for `BTreeSet<u8>`
          but trait `From<[u8; _]>` is implemented for it
  = help: for that trait implementation, expected `[u8; _]`, found `Value`
  = note: required for `Value` to implement `Into<BTreeSet<u8>>`
  = note: required for `BTreeSet<u8>` to implement `TryFrom<Value>`
  = note: required for `Value` to implement `TryInto<BTreeSet<u8>>`
  = note: this error originates in the derive macro `ZeekType` (in Nightly builds, run with -Z macro-backtrace for more info)
